### DISCUSSION TITLE
Supported bit depths

### DISCUSSION THREAD

**Discussion by BitPerfectAudio** [ExistentialAudio/BlackHole#42] [Feature Requests]
It would be beneficial to support integer 16 and 24bit, in addition to 32 float

**Comment by devinroth** [comment]
I'll address this first. It's actually not beneficial to support 16 and 24 bit audio. On macOS the HAL runs everything using 32 bit float regardless of what programs write to the buffer. And 32 bit float is more than enough to support 16 and 24 bit integers without any loss of resolution with the added benefit that I don't have to worry about clipping. 

That being said, I could allow 16 and 24 bit audio formats but they will be converted into 32 bit float in the back end regardless. All DAWs (at least well programmed ones) are already doing all the math using float32 or float64. The question then becomes should that conversion happen on the application side or the driver side. 

The only downside to 32bit float is file size but that's not an issue for BlackHole.

I'll leave this up if someone else wants to chime in.

**Comment by BitPerfectAudio** [comment]
with "beneficial" I meant being nice with the application playing audio, so it doesn't try to change anything. I agree that a DAW will generally always use float32 anyhow but I was trying to make a different point, as the application using the audio might not be a DAW

**Comment by devinroth** [comment]
I see your point but i feel like it’s a pretty rare case that an application won’t support 32 bit float output. 

Anyway I’ll keep this up but it’s pretty low on my priorities. 

Devin

**Reply by aguitteau** [reply]
Hi, my technical knowledge of audio is limited so bear with me if I come across as a layman. I found this thread because I'm looking for a way to record audio off my computer in 24-bit as opposed to 32 (which is what Soundflower offers). I'd like this option because currently, if I drag a 32-bit audio file into Logic Logic will put a 24-bit converted version in the file's directory. This isn't a huge deal but it's inconvenient to have to delete that duplicate every time I put a file in, and with hundreds of files it adds up (especially when I'm trying to keep my folders clean). It would be nice if I could just record in 24 so I could skip that step.

**Reply by devinroth** [reply]
Seems more likely that BlackHole will add 24 bit than Logic supporting 32-bit audio eh? I've also ran into that annoyance with Logic. Make sure you also put in a feature request with Logic.

**Reply by devinroth** [reply]
I do have it on the list but it's low priority.

**Comment by xster** [comment]
I have another banal reason which is that it would be one way to get Rocksmith the game to work with a normal interface through a virtual channel instead of using their hardware dongle. 

https://www.reddit.com/r/rocksmith/comments/5uwunl/so_i_just_made_rocksmith_work_without_the/

But the game only recognizes 16 bit devices.

**Comment by devinroth** [comment]
Hahaha. That’s so stupid but a legit reason.
